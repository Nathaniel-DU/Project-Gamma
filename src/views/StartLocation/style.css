h2 {
    color: white;
    text-align: center;
    font-family: 'Sen', sans-serif;
    /* margin-top: 200px; */
}

.start-button {
    width: 50%;
    margin-top: 50px;
    display: block;
    margin: auto;
    font-family: 'Sen', sans-serif;
}

h1 {
    color: #ADEFD1FF;
    display: flex;
    justify-content: flex-end;
    margin-bottom: 100px!important;
    padding-right: 30px;
    padding-top: 25px;
    font-family: 'Sen', sans-serif;
}

.start-location {
    margin-bottom: 200px;
}

.start-trip {
    margin-top: 200px;
}

