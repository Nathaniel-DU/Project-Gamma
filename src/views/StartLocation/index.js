import React, { Fragment, Component } from "react";
import StartLocationButton from "../../components/StartLocationButton";
import UpdateView from "../UpdateView";
import Hamburger from '../../components/Hamburger';
import Header from '../../components/Header';
import './style.css';
import axios from 'axios';

export default class StartLocation extends Component {

  constructor() {
    super()

    this.state = {
      latitude: '',
      longitude: '',
      eventStarted: false
    }

    this.getMyLocation = this.getMyLocation.bind(this);
    this.startTrip = this.startTrip.bind(this);
  }

  getMyLocation() {
    const location = window.navigator && window.navigator.geolocation

    if (location) {
      location.getCurrentPosition((position) => {
        this.setState({
          latitude: position.coords.latitude,
          longitude: position.coords.longitude,
        })
        axios.post('/event/start', {
          lat: this.state.latitude,
          long: this.state.longitude,
        })
        .catch(err => console.log(err));
      }, (error) => {
        this.setState({ latitude: 'err-latitude', longitude: 'err-longitude' })
      });
    }
  }

  startTrip() {
    this.setState({ eventStarted: true });
  }

  render() {

    return (
      <div id="outer-container">
        <Hamburger />
        <main id="page-wrap">
          <Fragment>
            {!this.state.eventStarted &&
            <div>
              <h1>StaySafe</h1>
              <div className="start-location">
                <div className="container start-trip"><h2>Click to begin trip.</h2></div>
                  <StartLocationButton
                    text='Start'
                    getMyLocation={this.getMyLocation}
                    startTrip={this.startTrip} />
                </div>
                </div>
            }
            {this.state.eventStarted && <UpdateView />}
          </Fragment>
        </main>
      </div>
    )
  }
}