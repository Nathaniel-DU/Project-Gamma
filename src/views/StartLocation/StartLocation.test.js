import React from 'react';
import { shallow } from 'enzyme';
import StartLocation from './index';

describe('StartLocation', () => {
    it('should render correctly in "debug" mode', () => {
        const component = shallow(<StartLocation debug />);
        expect(component).toMatchSnapshot();
    })
})