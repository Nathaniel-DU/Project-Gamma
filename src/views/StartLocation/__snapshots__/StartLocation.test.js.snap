// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StartLocation should render correctly in "debug" mode 1`] = `
<div
  id="outer-container"
>
  <Hamburger />
  <main
    id="page-wrap"
  >
    <div>
      <h1>
        StaySafe
      </h1>
      <div
        className="start-location"
      >
        <div
          className="container start-trip"
        >
          <h2>
            Click to begin trip.
          </h2>
        </div>
        <StartLocationButton
          getMyLocation={[Function]}
          startTrip={[Function]}
          text="Start"
        />
      </div>
    </div>
  </main>
</div>
`;
